ClimsoftVer4/ClimsoftVer4/ucrValueFlagPeriod.vb

```diff
@@ -111,9 +112,56 @@ Public Class ucrValueFlagPeriod
         End If
     End Sub
 
+    Public Sub SetBackColor(backColor As Color)
+        ucrValue.SetBackColor(backColor)
+        ucrFlag.SetBackColor(backColor)
+        If bIncludePeriod Then
+            ucrPeriod.SetBackColor(backColor)
+        End If
+    End Sub
+
+    Public Overrides Sub SetValue(objNewValue As Object)
+        Dim lstValueFlagPeriod As List(Of Object)
+
+        MyBase.SetValue(objNewValue)
+        lstValueFlagPeriod = TryCast(objNewValue, List(Of Object))
+        'TODO
+        'Not sure about this check. No certain about whether we should force the
+        'developer to always pass a list with a min of 2 values for Value and flag
+        If lstValueFlagPeriod.Count = 3 Then
+            ucrValue.SetValue(lstValueFlagPeriod(0))
+            ucrFlag.SetValue(lstValueFlagPeriod(1))
+            If bIncludePeriod Then
+                ucrPeriod.SetValue(lstValueFlagPeriod(2))
+            End If
+        ElseIf lstValueFlagPeriod.Count = 2 Then
+            ucrValue.SetValue(lstValueFlagPeriod(0))
+            ucrFlag.SetValue(lstValueFlagPeriod(1))
+        ElseIf lstValueFlagPeriod.Count = 1 Then
+            ucrValue.SetValue(lstValueFlagPeriod(0))
+        End If
+    End Sub
+
+    ''' <summary>
+    ''' Gets the value for ucrValue control
+    ''' </summary>
+    ''' <returns></returns>
+    Public Function GetValueValue() As String
+        Return ucrValue.GetValue
+    End Function
+
+    Public Function IsValueValueEmpty() As Boolean
```

Comment:
`ValueValue` sounds strange?

Reply:
Any Suggestion? That would easily suggest that the function retrieves value for ucrValue

Reply:
EntryValue or ElementValue something like that maybe.